pre_commit/meta_hooks/check_useless_excludes.py

```diff
@@ -0,0 +1,41 @@
+import re
+import sys
+
+import pre_commit.constants as C
+from pre_commit.clientlib import load_config
+from pre_commit.git import get_all_files
+
+
+def exclude_matches_any(filenames, include, exclude):
+    include_re, exclude_re = re.compile(include), re.compile(exclude)
+    for filename in filenames:
+        if include_re.search(filename) and exclude_re.search(filename):
+            return True
+    return False
+
+
+def check_useless_excludes(config_file=None):
+    config = load_config(config_file or C.CONFIG_FILE)
+    files = get_all_files()
+    useless_excludes = False
+
+    exclude = config.get('exclude')
+    if exclude != '^$' and not exclude_matches_any(files, '', exclude):
+        print('The global exclude pattern does not match any files')
+        useless_excludes = True
+
+    for repo in config['repos']:
+        for hook in repo['hooks']:
+            include, exclude = hook.get('files', ''), hook.get('exclude')
```

Comment:
hmmm, unless I'm thinking about this wrong I don't think `files` is relevant to this hook?

Reply:
This way there will be an error when you're excluding a file that exists, but is not included by `files`.

For example, you could have a hook that only runs on Python files, but has an `exclude` pattern for the README. Without looking at `files`, `check-useless-excludes` would conclude that everything is fine, although in reality the `exclude` is never matched.

Reply:
oh nice, great idea!